spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/RDDFunctions.scala

```diff
@@ -92,7 +92,9 @@ class RDDFunctions[T](rdd: RDD[T]) extends WritableToCassandra[T] with Serializa
     rwf: RowWriterFactory[T],
     columnMapper: ColumnMapper[T]): Unit = {
 
-    val table = TableDef.fromType[T](keyspaceName, tableName)
+    val protocolVersion = connector.withClusterDo(_.getConfiguration.getProtocolOptions.getProtocolVersion)
```

Comment:
This is the "actual" protocol version being used. The Configuration entry is for forcing an alternate PV. The Default "null" uses the "HighestSupportedVersion" by the driver.

Connecting to C* 3.X with the 3.0 Driver and the Parameter == null yields PV4
Connecting to C* 3.X with the 3.0 Driver and the Parameter == PV3 yields PV3
Connecting to C* 2.1.X with the 3.0 Driver and the Parameter == null yields PV3

I can remove this now though as I decided to just gate the whole test suite